client/my-sites/domains/index.js

```diff
@@ -98,10 +103,24 @@ export default function () {
 		domainManagementController.domainManagementRedirectSettings
 	);
 
-	registerStandardDomainManagementPages(
-		paths.domainManagementEditContactInfo,
-		domainManagementController.domainManagementEditContactInfo
-	);
+	if ( isAllDomainManagementEnabled ) {
+		page(
+			paths.domainManagementAllRoot() + '/:domain/edit-contact-info/:site',
+			siteSelection,
+			navigation,
+			domainManagementController.domainManagementSubpageParams( EDIT_CONTACT_INFO ),
+			domainManagementController.domainManagementEditContactInfo,
+			domainManagementController.domainManagementSubpageView,
+			domainManagementController.domainDashboardLayout,
+			makeLayout,
+			clientRender
+		);
+	} else {
+		registerStandardDomainManagementPages(
+			paths.domainManagementEditContactInfo,
+			domainManagementController.domainManagementEditContactInfo
+		);
+	}
```

Comment:
I think this'll make the older page retire once this flag is on, right? In case that's correct, I think we should change the logic because both pages will co-exist. More on it here - https://a8c.slack.com/archives/C082KU0C0MR/p1734032282145499?thread_ts=1734026453.536669&cid=C082KU0C0MR